site/src/components/Resources/ResourceAvatar.tsx

```diff
@@ -22,15 +24,13 @@ const iconByResource: Record<string, typeof MemoryIcon> = {
   google_compute_instance: AdjustedMemoryIcon,
   aws_instance: AdjustedMemoryIcon,
   kubernetes_deployment: AdjustedMemoryIcon,
-  null_resource: HelpIcon,
+  null_resource: WidgetsIcon,
 }
 
-export type ResourceAvatarProps = { type: string }
+export type ResourceAvatarProps = { type: ValidIcons }
 
 export const ResourceAvatar: React.FC<ResourceAvatarProps> = ({ type }) => {
-  // this resource can return undefined
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-  const IconComponent = iconByResource[type] ?? HelpIcon
+  const IconComponent = iconByResource[type] ?? WidgetsIcon
```

Comment:
Since the resource can be `undefined`, I modified the return type of `iconByResource` to match that, which lets us remove this `eslint` line.